ferrocene/tools/self-test/src/linkers.rs

```diff
@@ -99,6 +187,249 @@ pub(crate) fn check_and_add_rustflags(
     Ok(())
 }
 
+/// Look at the arguments given to the linker.
+///
+/// * Returns `Ok(true)` if these linker arguments look OK
+/// * Returns `Ok(false)` if the linker arguments were not OK, but an extra
+///   argument was added to `compiler_args` and so the caller should try again
+/// * Returns `Err(...)` if the linker arguments were not OK, and there's no
+///   proposed remedy
+fn linker_args_ok<'a, I>(
+    target: &str,
+    linker_args: I,
+    compiler_args: &mut Vec<String>,
+) -> Result<bool, Error>
+where
+    I: Iterator<Item = &'a str>,
+{
+    let linker_args = rationalise_linker_args(linker_args);
+    for arg in linker_args {
+        match arg {
+            LinkerArg::Input(_) => {}
+            LinkerArg::Output(_) => {}
+            LinkerArg::LibraryPath(_) => {}
+            LinkerArg::DiscardAll => {}
+            LinkerArg::Keyword(_) => {}
+            LinkerArg::Link(_) => {}
+            LinkerArg::Emulation(_) => {}
+            LinkerArg::PluginOpt(_) => {}
```

Comment:
Interesting. I don't think plugin options have any effect if there is no plugin loaded because they only apply to the previously named plugin, so I think they are 'safe' (citing https://gcc.gnu.org/wiki/whopr/driver because the [binutils LD page](https://sourceware.org/binutils/docs/ld/Options.html) doesn't mention -plugin-opt).

Reply:
Probably, I'd feel better if we just disallow it though :sweat_smile:

Reply:
I'll think about how to do this.

Reply:
Turns out it was easy. If you see a -plugin-opt then you definitely haven't seen a -plugin, because -plugin always breaks us out of the loop.